### PR TITLE
add option to configure pre- and post-testtasks per test

### PR DESCRIPTION
**Pull request by RuepInc**
This PR adds 2 new config options:

`preTestTasks` and `postTestTasks`

With these options the execution of `RunnerScripts`, `Runners` and `DatabaseScripts` can be toggled and ordered. 
If not provided the default order is applied.

### CODE REVIEW

**Review by rubengees — APPROVED** [review]
Nice, looks good! :clap: 

Some minor comments regarding style, but otherwise good to go.

**Comment by rubengees** [comment]
I just noticed: We should also extend the README with the new config options.

File: src/main/kotlin/de/smartsquare/squit/config/ConfigExtensions.kt [hunk 1]
```diff
@@ -104,6 +108,20 @@ val Config.preRunners get() = getSafeStringList(PRE_RUNNERS)
...
+ */
+val Config.preTestTasks get() =
+    when (hasPath(PRE_TEST_TASKS)) {
+        true -> getEnumList(SquitPreTestTask::class.java, PRE_TEST_TASKS)!!
+        else -> listOf(
+            SquitPreTestTask.PRE_RUNNERS,
+            SquitPreTestTask.PRE_RUNNER_SCRIPTS,
+            SquitPreTestTask.DATABASE_SCRIPTS
+        )
+    }
```
[the remark this answers is not in the document]
Comment (RuepInc) [reply src/main/kotlin/de/smartsquare/squit/config/ConfigExtensions.kt]: Good point. Added the type explicitly

File: src/main/kotlin/de/smartsquare/squit/task/SquitRequestTask.kt [hunk 2]
```diff
@@ -58,6 +60,7 @@ import java.util.concurrent.TimeUnit
  * Task for running requests against the given api. Also capable of running existing sql scripts before and after the
  * request.
  */
+@Suppress("TooManyFunctions")
```
[the remark this answers is not in the document]
Comment (RuepInc) [reply src/main/kotlin/de/smartsquare/squit/task/SquitRequestTask.kt]: Done

File: src/test/kotlin/de/smartsquare/squit/config/ConfigExtensionsTest.kt [hunk 3]
```diff
@@ -491,4 +492,35 @@ class ConfigExtensionsTest {
...
+    fun `config object with an invalid preTestTask`() {
+        val config = ConfigFactory.parseMap(
+            mapOf(
+                "endpoint" to "https://example.com",
+                "preTestTasks" to listOf("[NotExistingTask]"),
+            )
+        )
+
+        val call = { config.validate() }
+        @Suppress("MaxLineLength")
```
[the remark this answers is not in the document]
Comment (RuepInc) [reply src/test/kotlin/de/smartsquare/squit/config/ConfigExtensionsTest.kt]: Removed

File: src/test/kotlin/de/smartsquare/squit/config/ConfigExtensionsTest.kt
[the remark this answers is not in the document]
Comment (RuepInc) [reply src/test/kotlin/de/smartsquare/squit/config/ConfigExtensionsTest.kt:1]: Added

File: README.md [hunk 4]
```diff
@@ -123,25 +123,27 @@ or `local.conf`) files.
...
+| postRunners            | An array of post runner classes to use.                                                                                                            | `postRunners = ["com.example.ExamplePostRunner"]`                             |
+| preRunnerScripts       | An array of paths to groovy pre runner scripts to use.                                                                                             | `preRunnerScripts = [./scripts/pre_runner.groovy]`                            |
+| postRunnerScripts      | An array of paths to groovy post runner scripts to use.                                                                                            | `postRunnerScripts = [./scripts/post_runner.groovy]`                          |
+| preTestTasks           | An array of tasks to execute after the test. The order of the entries determines the order of execution.                                           | `postTestTasks = ["DATABASE_SCRIPTS", "POST_RUNNERS", "POST_RUNNER_SCRIPTS"]` |
```
Comment (rubengees) [inline README.md:143]: Typo I think

```suggestion
| postTestTasks           | An array of tasks to execute after the test. The order of the entries determines the order of execution.                                           | `postTestTasks = ["DATABASE_SCRIPTS", "POST_RUNNERS", "POST_RUNNER_SCRIPTS"]` |
```

**Review by rubengees — APPROVED** [review]
:ok_hand:

**Pull request merged**